src/scratch/ScratchRuntime.as

```diff
@@ -77,8 +85,56 @@ public class ScratchRuntime {
 			saveAfterInstall = false;
 			return;
 		}
-
-		if (recording) saveFrame(); // Recording a YouTube video?  Old / Unused currently.
+		if (ready==0) {
+			var tR:Number = getTimer()*.001-secs;
+			while (t>sounds.length/framerate+1/framerate) {
+				saveSound();
+			}
+			var count:int = 3;
+			if (tR>=3.75){
+				ready = 1;
+				count = 1;
+				sounds = [];
+				frames=[];
+				Scratch.app.log(LogLevel.TRACK, "Project video started");
+			}
+			else if (tR>=2.5){
+				count=1
+			}
+			else if (tR>=1.25 && mReady) {
+				count=2;
+			}
+			else if (tR>=1.25) {
+				secs+=tR;
+			}
+			else {
+				app.refreshStagePart();
+			}
+		}
+		if (recording) { // Recording a YouTube video?
```

Comment:
Could you explain what this code section is doing? This is the way I'm currently reading it, some of which I likely guessed wrong :)
```
if (the recording has been going for too long) {
    stop recording but grab this last frame
}
else { // usual case during recording
    if (we haven't fed this frame to the encoder yet, and the frame number is divisible by 2 or 3) {
        feed this frame and its sound data to the encoder
        forget this frame and its sound data
    }
    if (we haven't fed this frame to the encoder yet, and the framerate is 30) {
        feed this frame and its sound data to the encoder
        forget this frame and its sound data
    }
```
If we're not recording at 30fps, do we hold onto frames indefinitely if they aren't divisible by 2 or 3? And by "indefinitely" I guess I mean "until starting a new recording", which seems to be when `clearRecording()` is called.